### PR TITLE
chore: add DI and tests for 6 zero-coverage pipeline functions

### PR DESCRIPTION
**Pull request by yvonnedevlinrh**
## Summary

Reduce CRAPload from 38 to 32 by adding dependency injection and unit tests to 6 functions in `internal/crap/` and `internal/aireport/` that previously had 0% line coverage. This creates an 8-function buffer below the CI threshold
(`--max-crapload=38`), addressing the fragility described in #166.

### Problem

CRAPload sat at exactly 38/38 (the CI threshold). Six pipeline orchestration functions had 0% coverage because their only tests were guarded by `testing.Short()`, which is skipped during `gaze crap`'s internal coverage generation (`go test -short -coverprofile=...`).

### Solution

- Added `contractCoverageDeps` struct to `internal/crap/contract.go` — injectable dependencies for `analyzePackageCoverage` (absorbs existing `aiMapperFn` variadic)
- Added `qualityPipelineDeps` struct to `internal/aireport/runner_steps.go` — injectable dependencies for `runQualityStep`, `runQualityForPackage`, `runClassifyStep`
- Added 31 new unit tests (10 in `crap`, 21 in `aireport`) — none guarded by `testing.Short()` so they contribute real coverage to CRAPload
- Updated `pipelineStepFuncs` signatures and test fakes in `pipeline_internal_test.go`
- Follows the established `pipelineStepFuncs` nil-means-default DI pattern

### Functions covered

| Function | Package | CRAP Before | CRAP After |
|----------|---------|------------|------------|
| `analyzePackageCoverage` | `crap` | 110.0 | 14.7 |
| `runQualityForPackage` | `aireport` | 110.0 | 10.0 |
| `loadTestPackage` | `crap` | 72.0 | < 15 |
| `runQualityStep` | `aireport` | 72.0 | < 15 |
| `runClassifyStep` | `aireport` | 72.0 | < 15 |
| `loadTestPackageForQuality` | `aireport` | 30.0 | < 15 |

### Files changed

- `internal/crap/contract.go` — DI struct + signature change (+54 lines)
- `internal/crap/contract_test.go` — 10 new tests (+225 lines)
- `internal/aireport/runner_steps.go` — DI struct + 3 signature changes (+90 lines)
- `internal/aireport/runner_steps_test.go` — 21 new tests (+456 lines)
- `internal/aireport/runner.go` — `pipelineStepFuncs` type updates (+4/-4 lines)
- `internal/aireport/pipeline_internal_test.go` — fake closure signature updates (+10/-10 lines)
- `openspec/changes/crapload-di-coverage-pr1a/` — spec artifacts

### Verification

- `go build ./...` — PASS
- `go test -race -count=1 -short ./...` — PASS (all 16 packages)
- `golangci-lint run` — 0 issues in changed files
- CRAPload: **38 → 32** (target met)

**RE: MEDIUM — duplicate DI resolution logic**
Acknowledged. The duplication between `contractCoverageDeps` (crap) and `qualityPipelineDeps` (aireport) is intentional — the `classifyResults` signatures genuinely differ between packages (design decision D2). 
Consolidation of `loadTestPackage`/`loadTestPackageForQuality` and related dedup is planned for PR 2a (`BuildContractCoverageFunc` decomposition).

Partially addresses #166 (Phase 1a).

### CODE REVIEW

File: openspec/changes/crapload-di-coverage-pr1a/proposal.md [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+- `runQualityForPackage`: Accepts optional injectable dependencies for testing
+- `runQualityStep`: Accepts optional injectable dependencies for testing
+- `runClassifyStep`: Accepts optional injectable dependencies for testing
+
+### Removed Capabilities
+- None
+
+## Impact
+
+- **Files modified**: `internal/crap/contract.go`, `internal/aireport/runner_steps.go`
```
Comment (em-redhat) [inline openspec/changes/crapload-di-coverage-pr1a/proposal.md]: **[MEDIUM] Spec file path drift**

The proposal, spec (`specs/di-coverage.md`), tasks, and PR description reference `internal/crap/contract.go` and `internal/crap/contract_test.go` throughout. The actual implementation is at `internal/provider/goprovider/contract.go` and `internal/provider/goprovider/contract_internal_test.go`.

Please update the file path references in all four spec artifacts and the PR description to match the real locations.

**Review by em-redhat — COMMENTED** [review]
## PR Review: #183 — chore: add DI and tests for 6 zero-coverage pipeline functions

Well-executed internal testability improvement. All 5 CI checks pass, no security concerns, constitution compliance is clean. The DI pattern follows the established `pipelineStepFuncs` nil-means-default approach, and the 31 new tests achieve the CRAPload reduction target (38 → 32).

One MEDIUM finding to address before I can approve:

**[MEDIUM] Spec file path drift**: The OpenSpec proposal, spec, tasks, and PR description all reference `internal/crap/contract.go` and `internal/crap/contract_test.go`, but the actual implementation is in `internal/provider/goprovider/contract.go` and `internal/provider/goprovider/contract_internal_test.go`. Please update the spec artifacts and PR description to match the real file paths.

Two LOW notes (non-blocking):
- Test naming uses `_DI_` as a category marker (e.g., `TestRunQualityForPackage_DI_Success`) — slightly unconventional per TC-003 but readable. No action needed.
- Several error-path test groups could be consolidated into table-driven tests per TC-006. Optional improvement.

_This review was generated by /review-pr (AI-assisted)._

File: openspec/changes/crapload-di-coverage-pr1a/proposal.md (hunk 1 above)
Comment (yvonnedevlinrh) [reply openspec/changes/crapload-di-coverage-pr1a/proposal.md]: Addressed in 0a1f661: Updated all file path references across 4 spec artifacts (proposal.md, design.md, specs/di-coverage.md, tasks.md) from internal/crap/contract.go to internal/provider/goprovider/contract.go and from internal/crap/contract_test.go to internal/provider/goprovider/contract_internal_test.go. 9 references corrected.

**Review by em-redhat — APPROVED** [review]
lgtm
